src/librustc_ast/ast.rs

```diff
@@ -1165,7 +1165,9 @@ pub enum ExprKind {
     /// and the remaining elements are the rest of the arguments.
     /// Thus, `x.foo::<Bar, Baz>(a, b, c, d)` is represented as
     /// `ExprKind::MethodCall(PathSegment { foo, [Bar, Baz] }, [x, a, b, c, d])`.
-    MethodCall(PathSegment, Vec<P<Expr>>),
+    /// This `Span` is the span of the function, without the dot and receiver
+    /// (e.g. `foo(a, b)` in `x.foo(a, b)`
+    MethodCall(PathSegment, Vec<P<Expr>>, Span),
```

Comment:
Is this `Span` not available as part of the `Ident` that is the name of the method, in `PathSegment`?

Reply:
This `Span` also includes the opening and closing parenthesis, so the `PathSegment` and `Vec<P<Expr>>` are insufficient to reconstruct it exactly.